omegaconf/basecontainer.py

```diff
@@ -328,13 +328,12 @@ def expand(node: Container) -> None:
                     dest[key] = target_node
                     dest_node = dest._get_node(key)
 
-            if (
-                dest_node is None
-                and is_structured_config(dest._metadata.element_type)
-                and not missing_src_value
-            ):
+            is_optional, et = _resolve_optional(dest._metadata.element_type)
+            if dest_node is None and is_structured_config(et) and not missing_src_value:
                 # merging into a new node. Use element_type as a base
-                dest[key] = DictConfig(content=dest._metadata.element_type, parent=dest)
+                dest[key] = DictConfig(
+                    et, parent=dest, ref_type=et, is_optional=is_optional
```

Comment:
More of a meta question for @omry: should a `DictConfig` created from a structured config type automatically set its `ref_type` to that type? (here, that would mean not needing to specify `ref_type=et`)

Reply:
> In the context of merge, I think the ref_type should be preserved from the target node.

Agreed. This PR is consistent with preserving ref_type from the target node.
> If there is no target node (new node), I think the ref type should be Any (default, irrc).

Usually, but I think that if parent's `element_type` is set
then (for a new node) we should have `parent_element_type == child_ref_type`.
For example:
```python
from omegaconf import OmegaConf, DictConfig, ValidationError
from tests import User
from typing import Any
from pytest import raises

cfg = OmegaConf.merge(DictConfig({}, element_type=User), {"a": User("Bond")})

# On this branch:
assert cfg.a._metadata.ref_type == User

# On master branch:
assert cfg.a._metadata.ref_type == Any
```
This change is consistent with how assignment works: `child_ref_type` is
inherited from `parent_element_type`.
```python
cfg2 = DictConfig({}, element_type=User)
cfg2.a = User("Bond")
# On this branch AND master branch:
assert cfg2.a._metadata.ref_type == User
```